src/coreclr/vm/threads.h

```diff
@@ -1764,23 +1777,31 @@ class Thread
     //--------------------------------------------------------------
     // Failable initialization occurs here.
     //--------------------------------------------------------------
-    BOOL InitThread();
+    void InitThread();
     BOOL AllocHandles();
 
     //--------------------------------------------------------------
     // If the thread was setup through SetupUnstartedThread, rather
     // than SetupThread, complete the setup here when the thread is
     // actually running.
-    // WARNING : only GC calls this with bRequiresTSL set to FALSE.
     //--------------------------------------------------------------
-    BOOL HasStarted(BOOL bRequiresTSL=TRUE);
+    BOOL HasStarted();
 
     // We don't want ::CreateThread() calls scattered throughout the source.
     // Create all new threads here.  The thread is created as suspended, so
     // you must ::ResumeThread to kick it off.  It is guaranteed to create the
     // thread, or throw.
     BOOL CreateNewThread(SIZE_T stackSize, LPTHREAD_START_ROUTINE start, void *args, LPCWSTR pName=NULL);
 
+private:
+    bool m_managedInitializationPerformed;
```

Comment:
This flag does not seem to be really needed. The managed side has it duplicated.

Reply:
This is for the specific native call not the managed logic. I appreciate that presently this call is restricted to only a managed call so could be handled on that end. My intent here is to avoid any consideration of making a managed call if the init hasn't been called. Given the minor size increase of the costly to activate `Thread` class I feel this is an acceptable trade off. Let me know if you think otherwise.

Reply:
> My intent here is to avoid any consideration of making a managed call if the init hasn't been called.

Is this logic ever going to kick in? I believe that the managed call is always only made exactly once in with the current implementation.

Reply:
>Is this logic ever going to kick in?

Yes since it isn't checked to avoid re-initialization but rather to permit always calling `Thread::CleanUpForManagedThreadInNative()`.

Reply:
Neither of the two places that call `Thread::CleanUpForManagedThreadInNative` today need this check. Both of them call `Thread::CleanUpForManagedThreadInNative` only once the initialization has been done. The finalizer case has an extra line to guarantee that instead of always calling `Thread::CleanUpForManagedThreadInNative`.

Reply:
Okay.